gramps/plugins/db/dbapi/dbapi.py

```diff
@@ -1183,3 +1185,43 @@ def _sql_cast_list(self, values):
         in the appropriate type.
         """
         return [v if not isinstance(v, bool) else int(v) for v in values]
+
+    def select_from_table(
+        self, table_name, what=None, where=None, order_by=None, env=None
+    ):
+        # DB-API implementation
+        # NOTE: evaluator takes a pattern in case your DB-API
+        #       varies in syntax for JSON extraction
+        evaluator = Evaluator(
+            table_name, 'json_data->>"$.{attr}"', env if env is not None else globals()
+        )
+
+        if what is None:
+            what_expr = "json_data"
+        elif isinstance(what, str):
+            what_expr = str(evaluator.convert(what))
+        else:
+            what_expr = ", ".join([str(evaluator.convert(w)) for w in what])
+
+        if where is None:
+            where_expr = ""
+        else:
+            where_expr = " WHERE %s" % evaluator.convert(where)
+
+        if order_by is None:
+            order_by_expr = ""
+        else:
+            order_by_expr = evaluator.get_order_by(order_by)
+
+        self.dbapi.execute(
+            f"SELECT {what_expr} from {table_name}{where_expr}{order_by_expr};"
```

Comment:
This looks very dangerous to me. `table_name="person; DROP TABLE person; SELECT * FROM person"` :boom: 

Of course you could say we leave the responsibility of sanitization to the downstream code like addons etc., but I would very much prefer if we could avoid template strings as much as possible!

Or am I missing something?

Reply:
Yes, I think you are missing something :) All of the expressions are parsed, and only specific items can be evaluated.

Reply:
Thanks, can you elaborate? Where is `table_name` parsed?

Reply:
>  I'll add an underscore to the name to make that more clear.

Yes, that's perfect, thanks!